pkg/plugins/generic/generic_plugin.go

```diff
@@ -124,6 +124,9 @@ func (p *GenericPlugin) Apply() error {
 	}
 
 	if err := utils.SyncNodeState(p.DesireState, pfsToSkip); err != nil {
+		if strings.Contains(err.Error(), "cannot allocate memory") {
+			p.AddToDesiredKernelParams(utils.KernelParamPciRealloc)
+		}
```

Comment:
iiuc, we might reboot twice?

Reply:
Yes it is possible to reboot multiple times depending on when AddToDesiredKernelParams gets a unique kernel param.